tests/AbstractTestCase.php

```
@@ -25,13 +28,31 @@ protected function setUp(): void
+        // Skip migrate:fresh when RefreshDatabase is used — that trait already
+        // handles migrations itself (via its own setUp hook, which already ran
+        // by the time we get here), and running migrate:fresh again inside the
+        // transaction it starts would be redundant and unsafe.
+        $usesRefreshDatabase = in_array(
+            \Illuminate\Foundation\Testing\RefreshDatabase::class,
+            array_keys((function () {
+                return class_uses_recursive($this);
+            })->call($this))
+        );
+
+        if ( ! $usesRefreshDatabase && ! Schema::hasTable('users')) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Illuminate\Support\Facades\Schema' in method 'setUp'. (undefined)

(StaticAccess)